### PR TITLE
Make it compatible with `gulp-reporter` & `gulp-eslint`

### PR DESCRIPTION
**Pull request by gucong3000**
- Compatible with `gulp-reporter`, easy to assign errors by developer
- Compatible with `gulp-eslint`, #14

### CODE REVIEW

File: readme.md [hunk 1]
```diff
@@ -34,24 +35,29 @@ gulp.task('default', () =>
...
 
-Report errors only.
+### [xo.format(formatter, output)](https://github.com/adametry/gulp-eslint/#eslintformatformatter-output)
+
+### [xo.formatEach(formatter, output)](https://github.com/adametry/gulp-eslint/#eslintformateachformatter-output)
+
+## Related
 
+- [gulp-eslint](https://github.com/adametry/gulp-eslint)
+- [gulp-reporter](https://github.com/gucong3000/gulp-reporter)
```
Comment (sindresorhus) [inline readme.md]: These need a description.

File: test.js [hunk 2]
```diff
@@ -1,24 +1,17 @@
...
-			hooker.unhook(gutil, 'log');
-			t.pass();
-		}
+	stream.on('data', file => {
+		t.truthy(file.eslint);
+		t.truthy(file.eslint.messages.length);
+		t.pass();
+	});
+	stream.on('error', () => {
+		t.fail();
```
Comment (sindresorhus) [inline test.js]: Use `p-event` here. See: https://github.com/sindresorhus/gulp-debug/blob/4db5871594742a346d17aa9b34f43c87d4e54934/test.js#L42-L44

File: readme.md [hunk 3]
```diff
@@ -34,24 +35,29 @@ gulp.task('default', () =>
...
+### [xo.failOnError()](https://github.com/adametry/gulp-eslint/#eslintfailonerror)
 
-Type: `boolean`<br>
-Default: `false`
+### [xo.failAfterError()](https://github.com/adametry/gulp-eslint/#eslintfailaftererror)
 
-Report errors only.
+### [xo.format(formatter, output)](https://github.com/adametry/gulp-eslint/#eslintformatformatter-output)
+
+### [xo.formatEach(formatter, output)](https://github.com/adametry/gulp-eslint/#eslintformateachformatter-output)
```
Comment (sindresorhus) [inline readme.md]: These should be sorted by usefulness:

- format
- failAfterError
- failOnError
- formatEach
- result
- results

File: package.json [hunk 4]
```diff
@@ -13,7 +13,7 @@
     "node": ">=4"
   },
   "scripts": {
-    "test": "xo --ignore='_fixture.js' && ava"
+    "test": "xo --ignore _fixture.js && ava"
```
Comment (sindresorhus) [inline package.json]: Don't do unrelated changes.

**Comment by sindresorhus** [comment]
You also need to make it use https://github.com/sindresorhus/eslint-formatter-pretty as a formatter by default.

**Comment by gucong3000** [comment]
@sindresorhus 
All done.
